miso-web/src/main/webapp/scripts/hot_sample.js

```diff
@@ -440,6 +440,7 @@ HotTarget.sample = (function() {
             },
             pack : function(sam, flat, errorHandler) {
               if (!Utils.validation.isEmpty(flat.stainName)) {
+                sam.stain = {};
```

Comment:
Since the user can switch from stained to unstained, I think this should be:
```
sam.stain = Utils.array.findFirstOrNull(Utils.array.namePredicate(flat.stainName), Constants.stains);
```